### ISSUE TITLE
ECOS fails when solving second time

### ISSUE THREAD

**Issue by whiterabbitfollow** [cvxgrp/cvxpygen#45]
Hey,

I've written a simple shortest path problem with constraints. When I compile the program with ECOS, the program is able to find a solution the first time it's queried. However, **the second time** it fails, even if the **input parameters are the same**. It seems to work fine if I run with SCS. The program that I compile is presented below.

```
def compile_optimization_program(self):
    m, n, w = self.m, self.n, self.w
    self.centers = cp.Parameter((m * n, w), name="centers")
    self.radiis = cp.Parameter((m * n), name="radiis")
    self.p_start = cp.Parameter((w, ), name="p_start")
    self.p_end = cp.Parameter((w,), name="p_end")
    self.Ps_stacked = cp.Variable((m * n, w), name=f"Ps")
    objective = cp.norm(self.Ps_stacked[1:] - self.Ps_stacked[:-1], axis=1).sum()
    const = (
            [
                cp.norm(self.Ps_stacked - self.centers, axis=1) <= self.radiis
            ]
            +
            [
                self.Ps_stacked[i-1] == self.Ps_stacked[i] for i in range(n, n*m, n)
            ]
            +
            [
                self.Ps_stacked[0] == self.p_start,
                self.Ps_stacked[-1] == self.p_end
            ]
    )
    self.problem = cp.Problem(cp.Minimize(objective), constraints=const)
```
I'm running python 3.10.12 and cmake  3.29.0. 

Complete code:
[bug.zip](https://github.com/cvxgrp/cvxpygen/files/14834062/bug.zip)

**Comment by The-SS** [comment]
The problem can also be replicated using the [MPC.ipynb](https://github.com/cvxgrp/cvxpygen/blob/master/examples/MPC.ipynb) code.
Simply running this a second time
```
t0 = time.time()
val = prob.solve(method='CPG')
t1 = time.time()
print('\nCVXPYgen\nSolve time: %.3f ms' % (1000 * (t1 - t0)))
print('Objective function value: %.6f\n' % val)
```
results in wrong results:

```
CVXPY
Solve time: 8.555 ms
Objective function value: 134.723940


CVXPYgen (first run)
Solve time: 3.146 ms
Objective function value: 134.722740


CVXPYgen (second run)
Solve time: 4.994 ms
Objective function value: 7195.184237
```

**Comment by SteveDiamond** [comment]
@maxschaller this is quite an important issue could you look into it?

**Comment by maxschaller** [comment]
@whiterabbitfollow @The-SS @SteveDiamond this is fixed in the newest release `v0.3.3`. I was able to reproduce the error with your code and got a reasonable matplotlib figure after the fix.
